extensions/amp-drive-viewer/0.1/amp-drive-viewer.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds the Google Docs viewer
+ *
+ * Example:
+ * <code>
+ * <amp-drive-viewer
+ *   layout="fixed-height"
+ *   src="https://www.example.com/my-document.pdf"
+ *   height="600">
+ * </amp-drive-viewer>
+ * </code>
+ */
+
+import {isExperimentOn} from '../../../src/experiments';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export const TAG = 'amp-drive-viewer';
+
+const ATTRIBUTES_TO_PROPAGATE = [
+  'alt',
+  'title',
+];
+
+const GOOGLE_DOCS_EMBED_RE = /^https?:\/\/docs\.google\.com.+\/pub.*\??/;
```

Comment:
Consider making HTTPS-only or always replace to HTTPS in `getSrc_`.

Reply:
I'm not sure that's necessary. It looks like Google Docs redirects `http` to `https` so detecting both will allow both to display correctly and securely. These URLs will be copy-pasted by the user from the "File -> Publish to web" menu in Google Docs so `http` will be rare. Do we rewrite `http` urls to `https` in other places for embedded content?